File: python/src/argonguard/passwords/_legacy.py
"""Legacy 擴充點（SPEC §6.4）：verify-only、建構時注入不可變清單。

核心不出貨任何 legacy 演算法實作（SPEC L3）；文件提供完整範例（如 bcrypt）。
"""

from typing import Protocol


class LegacyPasswordVerifier(Protocol):
    """Legacy verifier 介面（SPEC L2）：``can_handle`` 為廉價前綴測試；``verify`` 做實際驗證。"""

    def can_handle(self, encoded_hash):
        # type: (str) -> bool
        """此 verifier 是否認領該編碼字串（廉價前綴測試）。"""
        ...

    def verify(self, password, encoded_hash):
        # type: (str, str) -> bool
        """驗證密碼是否符合該 legacy 雜湊。"""
        ...

File: core/src/legacy.ts
/**
 * Legacy 格式驗證擴充點（SPEC §6.4）。僅能於建構 hasher 時以不可變有序清單注入；
 * 執行期動態註冊在 API 形狀上不可能。核心不內建任何 legacy 演算法實作；bcrypt 等範例見文件。
 */
export interface LegacyPasswordVerifier {
  /** 廉價前綴判斷：此 verifier 是否認領該字串。不得執行昂貴運算。 */
  canHandle(encodedHash: string): boolean;

  /** 驗證密碼。僅在 canHandle 回 true 時被呼叫。可為同步或非同步（Node 慣例）。 */
  verify(password: string, encodedHash: string): boolean | Promise<boolean>;
}
